solr/core/src/test/org/apache/solr/search/QueryEqualityTest.java

```diff
@@ -1362,6 +1366,15 @@ protected void assertQueryEquals(
     }
   }
 
+  private static FunctionQuery unwrapFQValueSource(FunctionQuery q) {
+    ValueSource vs = q.getValueSource();
+    if (vs instanceof WrappedFieldValueSource) {
+      return new FunctionQuery(((WrappedFieldValueSource) vs).unwrap());
```

Comment:
I'm wondering, why FunctionQuery can't get along with wrapped instance? Isn't it functionally equivalent to the backing one?